src/fixtures/project.go

```diff
@@ -0,0 +1,220 @@
+	"keboola-as-code/src/client"
+	"keboola-as-code/src/fixtures/testEnv"
+	"keboola-as-code/src/model/remote"
+	"keboola-as-code/src/utils"
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+)
+
+type TestProject struct {
+	t              *testing.T
+	testDir        string
+	stateFilePath  string
+	stateFile      *StateFile
+	api            *api.StorageApi
+	logs           *utils.Writer
+	defaultBranch  *remote.Branch
+	branchesByName map[BranchName]*remote.Branch
+}
+
+func NewTestProject(t *testing.T, stateFilePath string) *TestProject {
+	_, testFile, _, _ := runtime.Caller(0)
+	testDir := filepath.Dir(testFile)
+	p := &TestProject{t: t, testDir: testDir, stateFilePath: stateFilePath, branchesByName: make(map[BranchName]*remote.Branch)}
+	p.createApi()
+	p.loadDefaultBranch()
+	p.loadStateFile()
+	p.log("Initializing test project \"%s\", id: \"%d\".", p.api.ProjectName(), p.api.ProjectId())
+	return p
+}
+
+// Clear deletes all project branches (except default) and all configurations
+func (p *TestProject) Clear() {
+	p.log("Clearing project ...")
+	branches := p.loadBranches()
+	pool := p.api.NewPool()
+	for _, branch := range branches {
+		if branch.IsDefault {
+			// Default branch cannot be deleted, so we have to delete all configurations
+			p.DeleteAllConfigs(pool, branch)
+		} else {
+			p.DeleteBranch(pool, branch)
+		}
+	}
+	if err := pool.StartAndWait(); err != nil {
+		assert.FailNow(p.t, fmt.Sprintf("cannot delete branches: %s", err))
+	}
+	p.log("Test project cleared.")
+}
+
+// InitState crates branches and configurations according stateFile
+func (p *TestProject) InitState() {
+	p.log("Setting project state ...")
+	p.CreateConfigsInDefaultBranch()
+	p.CreateBranches()
+	p.CreateConfigsInBranches()
+	p.log("Project state set.")
+}
+
+func (p *TestProject) DeleteBranch(pool *client.Pool, branch *remote.Branch) *client.Request {
+	if branch.IsDefault {
+		panic(fmt.Errorf("default branch cannot be deleted"))
+	}
+	return p.api.DeleteBranchRequest(branch.Id).Send()
+}
+
+func (p *TestProject) DeleteAllConfigs(pool *client.Pool, branch *remote.Branch) {
+	if !branch.IsDefault {
+		panic(fmt.Errorf("only configs from default branch can be deleted"))
+	}
+
+	pool.
+		Request(p.api.ListComponentsRequest(branch.Id)).
+		OnSuccess(func(response *client.Response) *client.Response {
+			for _, component := range *response.Result().(*[]*remote.Component) {
+				for _, config := range component.Configs {
+					// Delete each configuration in branch
+					p.DeleteConfig(pool, config.ComponentId, config.Id)
+				}
+			}
+			return response
+		}).
+		Send()
+}
+
+func (p *TestProject) DeleteConfig(pool *client.Pool, componentId string, configId string) *client.Request {
+	return pool.
+		Request(p.api.DeleteConfigRequest(componentId, configId)).
+		Send()
+}
+
+func (p *TestProject) CreateBranches() {
+	// Create branches sequentially, parallel requests don't work good with this endpoint
+	for _, branch := range p.stateFile.Branches {
```

Comment:
Na tomto som zabil 2h, ... ked spustit requesty na vytvorenie branche paralelne, ... tak uspesne dobehne iba 1 ... hm .. mozno bug v KBC, ... zatial som tu dal, ze sa to spusta postupne.

Reply:
Inde problem s paralelnými requestmi nie je.